test/Engine/ProtoTest/FFITests/ElementRewriterTests.cs

```diff
@@ -52,19 +126,177 @@ public void LookupResolvedName_FromCompiler_RewriteAst()
             Assert.AreEqual("FFITarget.ElementResolverTarget", resolvedName);
         }
 
-        // Add tests for ElementRewriter.GetClassIdentifiers() api
         [Test]
-        public void GetClassIdentifiers_FromAst()
+        public void LookupResolvedName_ForTypedIdentifierFromCompiler_RewriteAst()
         {
-            var astNodes = CoreUtils.BuildASTList(core, "d = Point.ByCoordinates(0, 0, pt.X + 5);");
-            var astNode = astNodes[0];
-
-            var elementRewriter = new ElementRewriter(null);
-            var identifiers = elementRewriter.GetClassIdentifiers(astNode);
-            var identifierListNodes = identifiers as IdentifierListNode[] ?? identifiers.ToArray();
-            Assert.AreEqual(2, identifierListNodes.Count());
-            Assert.AreEqual("pt.X", identifierListNodes.ElementAt(0).ToString());
-            Assert.AreEqual("Point.ByCoordinates(0, 0, (pt.X) + 5)", identifierListNodes.ElementAt(1).ToString());
+
+            const string code = @"import (""FFITarget.dll"");";
+            var mirror = thisTest.RunScriptSource(code);
+
+            var testCore = thisTest.GetTestCore();
+            var astNodes = CoreUtils.BuildASTList(testCore, "d : ElementResolverTarget;");
+
+            var elementResolver = new ElementResolver();
+
+            var newNodes = ElementRewriter.RewriteElementNames(testCore.ClassTable, elementResolver, astNodes);
+
+            Assert.AreEqual("d : FFITarget.ElementResolverTarget", newNodes.ElementAt(0).ToString());
+
+            // Add verification for contents of element resolver resolution map
+            var assembly = elementResolver.LookupAssemblyName("ElementResolverTarget");
+            var resolvedName = elementResolver.LookupResolvedName("ElementResolverTarget");
+
+            Assert.AreEqual("FFITarget.dll", assembly);
+            Assert.AreEqual("FFITarget.ElementResolverTarget", resolvedName);
         }
+
+        [Test]
+        public void LookupResolvedName_ForComplexExpression_RewriteAst()
+        {
+            var astNodes = CoreUtils.BuildASTList(core, "p = Point.ByCoordinates(x, y, z).X;");
+
+            var elementResolver = new ElementResolver();
+            elementResolver.AddToResolutionMap("Point", "Autodesk.DS.Geometry.Point", "Protogeometry.dll");
```

Comment:
Related to my prev comment on the resolver, how does this affect the classTable lookup?